plugins/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/util/ProjectFactory.java

```diff
@@ -142,25 +156,34 @@ public IProject createProject(IProgressMonitor monitor, Shell shell) {
 			project.setDefaultCharset(defaultCharset, subMonitor.newChild(1));
 			createFolders(project, subMonitor, shell);
 
-			// first run contributors...
-			if (contributors != null) {
-				IFileCreator fileCreator = new IFileCreator() {
+			IFileCreator fileCreator = new IFileCreator() {
+				@Override
+				public IFile writeToFile(CharSequence chars, String fileName) {
+					return ProjectFactory.this.writeToFile(chars, fileName, project, subMonitor);
+				}
+			};
 
-					@Override
-					public IFile writeToFile(CharSequence chars, String fileName) {
-						return ProjectFactory.this.writeToFile(chars, fileName, project, subMonitor);
-					}
-				};
-				for (IProjectFactoryContributor contributor : contributors) {
+			// first run early contributors...
+			// in some cases it is crucial to enhance the project
+			// only after contributions, like in the case of gradle projects
+			// see https://bugs.eclipse.org/bugs/show_bug.cgi?id=492728
+			if (earlyContributors != null) {
```

Comment:
Can we extract a protected method that accepts contributors and project to removed duplicate code here and below?